addons/payment/models/payment_acquirer.py

```diff
@@ -200,17 +239,40 @@ def _get_feature_support(self):
     @api.model
     def create(self, vals):
         image_resize_images(vals)
+        if not vals.get('display'):
+            payment_icon_ids = vals.get('payment_icon_ids')
+            if payment_icon_ids and len(payment_icon_ids[0]) >= 3 and payment_icon_ids[0][2]:
+                vals['display'] = "Credit Card (powered by {0})".format(vals['name'])
+            else:
+                vals['display'] = vals['name']
         return super(PaymentAcquirer, self).create(vals)
 
     @api.multi
     def write(self, vals):
         image_resize_images(vals)
-        return super(PaymentAcquirer, self).write(vals)
+        super(PaymentAcquirer, self).write(vals)
+        for acquirer in self:
+            vals = {}
+            if acquirer.missing_required_fields:
+                vals['website_published'] = False
+                super(PaymentAcquirer, self).write(vals)
 
     @api.multi
     def toggle_website_published(self):
-        self.write({'website_published': not self.website_published})
-        return True
+        for acquirer in self:
+            if not acquirer.missing_required_fields:
+                self.write({'website_published': not acquirer.website_published})
+                return True
+            raise ValidationError("Some credentials are missing to activate the payment acquirer: \n %s"
+                                  % acquirer.missing_required_fields)
+
+    @api.multi
+    def _get_website_publication_state(self):
```

Comment:
According to guideline, a compute method for the fields ``foo`` should be named ``_compute_foo``